xbmc/video/VideoDatabase.cpp

```diff
@@ -3571,6 +3599,87 @@ bool CVideoDatabase::GetStreamDetails(CVideoInfoTag& tag) const
   return retVal;
 }
  
+bool CVideoDatabase::GetStreamDetails(const std::string &media_type, std::map<int, CVideoInfoTag>& details)
+{
+  bool retVal = false;
+  std::string media_ids;
+
+  std::unique_ptr<Dataset> pDS(m_pDB->CreateDataset());
+  try
+  {
+    std::string id;
+    if (media_type == MediaTypeMovie)
+      id = "idMovie";
+    else if (media_type == MediaTypeEpisode)
+      id = "idEpisode";
+    else if (media_type == MediaTypeMusicVideo)
+      id = "idMVideo";
+    else
+      return false;
+
+    media_ids = StringUtils::Join(details);
+    std::string strSQL = PrepareSQL("SELECT %s.%s, streamdetails.* FROM streamdetails "
+                                    "JOIN %s ON %s.idFile = streamdetails.idFile "
+                                    "WHERE %s.%s IN (%s)", 
+                                    media_type.c_str(), id.c_str(), media_type.c_str(), media_type.c_str(), 
+                                    media_type.c_str(), id.c_str(), media_ids.c_str());
+    pDS->query(strSQL);
+
+    while (!pDS->eof())
+    {
+      CStreamDetail::StreamType e = (CStreamDetail::StreamType)pDS->fv(2).get_asInt();
+      switch (e)
+      {
+      case CStreamDetail::VIDEO:
```

Comment:
This whole logic of reading stream details from the database is almost identical to the one in the existing `GetStreamDetails()` except for the index in the result set. Could we move this into a helper function that takes the result set and an index offset and returns a `CStreamDetail` object?